### DISCUSSION TITLE
What does it do?

### DISCUSSION THREAD

**Discussion by omar-dulaimi** [MichalLytek/typegraphql-nestjs#31] [Q&A] [answered]
Hello,

I'm honestly lost as to what this library does or how to use it. I tried the examples but couldn't understand what's the difference between this and pure nestjs. I definitely don't know what i'm talking about. Could you explain a little bit?

**Comment by MichalLytek** [comment, marked as answer]
@omar-dulaimi NestJS has its own GraphQL implementation that mimics TypeGraphQL decorators.
`typegraphql-nestjs` allows you to use standard TypeGraphQL (with middlewares, authorization, etc.) in NestJS app.

**Reply by omar-dulaimi** [reply]
@MichalLytek Oh you mean the types like Field, InputType and stuff like that?
So it won't help in querying nested resolvers unless written manually, right?

**Reply by MichalLytek** [reply]
You are looking for autogenerated resolvers for TypeORM? Maybe Warthog will interest you:
https://github.com/goldcaddy77/warthog

**Reply by omar-dulaimi** [reply]
I have never heard of it before, thanks for letting me know about it. What I'm trying to achieve can be done with type graphql prisma, but right now I'm using nestjs so was looking for something similar.

**Reply by omar-dulaimi** [reply]
Do you recommend that I go back to using TypeGraphql Prisma and leave Nestjs?

**Reply by MichalLytek** [reply]
As an author of TypeGraphQL, I'm a bit biased 😜

**Reply by omar-dulaimi** [reply]
Well, to me TypeGraphQL is a lot better than Nestjs despite the features they provide. I also noticed that TypeGraphql is faster.
